brokers/che-plugin-broker/broker/broker.go

```diff
@@ -114,24 +115,34 @@ func (cheBroker *ChePluginBroker) processPlugin(meta model.PluginMeta) error {
 	if err != nil {
 		return err
 	}
+	var pluginPath string
 
-	archivePath := filepath.Join(workDir, "pluginArchive.tar.gz")
-	pluginPath := filepath.Join(workDir, "plugin")
-
-	// Download an archive
-	cheBroker.PrintDebug("Downloading archive '%s' for plugin '%s:%s' to '%s'", url, meta.ID, meta.Version, archivePath)
-	err = cheBroker.ioUtil.Download(url, archivePath)
-	if err != nil {
-		return err
-	}
+	if strings.HasSuffix(url, pluginFileName) {
```

Comment:
We have no obligation that URL should end up with a specific extension or name. And in some cases, web servers serve files, not under their real names or content can be REST API instead of a real file.
Not sure this case is so simple.

Reply:
@garagatyi So Do you propose apply file type field to detect what we have: file or archive? As for me for quick start - suffix solution could be, than we could improve it. What Do you think?

Reply:
@garagatyi We can change strategy and download first and then try to guess if it is an archive or not. Is that you are referring?

Reply:
@skabashnyuk yes, I'm referring to that. But I agree with @AndrienkoAleksandr that we can start from suffix matching. Even though it is not covering all the cases it is super easy to implement.
My concerns are:
- not sure somebody would take the hard part if we have this simple suffix matching
- if unarchiving is not needed than Che master could download plugin YAML itself and not run plugin brokering

At the same time, we don't yet know how exactly brokering refactoring would be done and when, so I'm +1 to go with the proposed implementation. It doesn't harm but can benefit UX a bit.